src/presentation/ui/page/mappings.templ

```diff
@@ -0,0 +1,338 @@
+                this.isCreateVhostModalOpen = false;
+            },
+            isDeleteVhostModalOpen: false,
+            openDeleteVhostModal(vhostHostname) {
+				this.resetPrimaryStates();
+
+				this.virtualHost.hostname = vhostHostname;
+				this.isDeleteVhostModalOpen = true;
+			},
+			closeDeleteVhostModal() {
+				this.isDeleteVhostModalOpen = false;
+			},
+			deleteVhostElement() {
+				htmx.ajax(
+					'DELETE',
+					'/api/v1/vhosts/' + this.virtualHost.hostname + '/',
+					{swap: 'none'},
+				);
+                this.removeAvailableVhostHostname();
+				this.closeDeleteVhostModal();
+			},
+            isCreateMappingModalOpen: false,
+            openCreateMappingModal() {
+				this.resetPrimaryStates();
+                this.resetAuxiliaryStates();
+
+                this.isCreateMappingModalOpen = true;
+            },
+            isCreateMappingFromVhostModalOpen: false,
+            openCreateMappingFromVhostModal(vhostHostname) {
+				this.resetPrimaryStates();
+                this.resetAuxiliaryStates();
+
+                this.virtualHost.hostname = vhostHostname
+				this.isCreateMappingFromVhostModalOpen = true;
+                this.isCreateMappingFromVhost = true;
+            },
+            closeCreateMappingModal() {
+                this.isCreateMappingModalOpen = false;
+                this.isCreateMappingFromVhostModalOpen = false;
+            },
+            isDeleteMappingModalOpen: false,
+            openDeleteMappingModal(mappingId, mappingPath) {
+				this.resetPrimaryStates();
+
+				this.mapping.id = mappingId;
+				this.mapping.path = mappingPath;
+				this.isDeleteMappingModalOpen = true;
+			},
+			closeDeleteMappingModal() {
+				this.isDeleteMappingModalOpen = false;
+			},
+			deleteMappingElement() {
+				htmx.ajax(
+					'DELETE',
+					'/api/v1/vhosts/mapping/' + this.mapping.id + '/',
+					{swap: 'none'},
+				);
+				this.closeDeleteMappingModal();
+			},
+        }))
+    })
+}
+
+templ MappingsIndex(vhostsWithMappings []dto.VirtualHostWithMappings) {
+	@MappingsIndexLocalState()
+	<div class="flex flex-col" x-data="mappings">
+		<div class="mb-6 flex flex-row items-center justify-between">
+			<div class="basis-[70%]">
+				@componentStructural.PageTitle(
+					"Mappings",
+					"Configure how the server handles different URLs and paths. You can specify where static files are located, set responses for HTTP status codes, redirect requests to different URLs, add inline HTML content, and adjust how the server interacts with services like PHP.",
+					"ph-graph")
+			</div>
+			<div class="my-4 flex space-x-5">
+				@componentForm.SubmitButton("Create virtual host", "ph-plus-square", "openCreateVhostModal()", false, false)
+				@componentForm.SubmitButton("Create mapping", "ph-plus-square", "openCreateMappingModal()", false, false)
+			</div>
+		</div>
+		<div id="mappings-tables" class="card w-full">
+			@MappingsRegularFormTable(vhostsWithMappings)
```

Comment:
Not that this is a problem, but why RegularForm if there is only one kind of table?

Reply:
You're right. Useless adjective removed!